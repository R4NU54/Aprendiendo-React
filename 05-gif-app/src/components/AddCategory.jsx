import { useState } from 'react';

// eslint-disable-next-line react/prop-types
export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('');

  const handleInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (!inputValue) return;
    setInputValue('');
    onNewCategory(inputValue);
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        type='text'
        placeholder='Search...'
        value={inputValue}
        onChange={handleInputChange}
      />
      {/* <button type='submit'>Search</button> */}
    </form>
  );
};
