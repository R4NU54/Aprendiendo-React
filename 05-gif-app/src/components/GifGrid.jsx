import { GiftItem } from './GiftItem';

import { useFetchGifs } from '../hooks/useFetchGifs';

// eslint-disable-next-line react/prop-types
export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <section className='section-search'>
      <h3>{category}</h3>
      {isLoading && <p>Loading...</p>}
      <div className='card-grid'>
        {images.map((images) => (
          <GiftItem
            key={images.id}
            {...images}
          />
        ))}
      </div>
    </section>
  );
};
