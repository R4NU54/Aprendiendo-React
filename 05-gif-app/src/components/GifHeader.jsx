import image from '../../GifApp.webp';

export const GifHeader = () => {
  return (
    <header className='header'>
      <img
        alt='GifApp'
        className='gif-app'
        src={image}
      />
      <h1>React Gif App</h1>
    </header>
  );
};
