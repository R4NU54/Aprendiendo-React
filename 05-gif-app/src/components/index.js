export * from './AddCategory';
export * from './GifGrid';
export * from './GifItem';
export * from './GifHeader.jsx';
