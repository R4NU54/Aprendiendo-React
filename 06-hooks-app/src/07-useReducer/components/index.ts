export * from './todoAdd';
export * from './todoList';
export * from './todoItem';
