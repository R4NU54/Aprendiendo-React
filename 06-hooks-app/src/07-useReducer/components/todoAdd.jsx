import { useForm } from '../../hooks/useForm';
export const TodoAdd = ({ onNewTodoAdd }) => {
  const { formState, handleChange, onResetForm, description } = useForm({
    description: '',
  });

  const handleSubmit = e => {
    e.preventDefault();
    if (description.length <= 1) return;

    const idTodo = crypto.randomUUID();

    const newTodo = {
      id: idTodo,
      description: description,
      done: false,
    };

    onNewTodoAdd(newTodo);
    onResetForm();
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        name="description"
        placeholder="Agrega todo..."
        autoComplete="off"
        className="form-control"
        value={description}
        onChange={handleChange}
      />
      <button type="submit" className="btn btn-outline-primary mt-1 btn-block">
        Agregar
      </button>
    </form>
  );
};
