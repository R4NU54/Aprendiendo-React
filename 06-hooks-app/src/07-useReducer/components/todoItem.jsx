export const TodoItem = ({ todo, onDeleteTodo, onDoneTodo }) => {
  return (
    <li className="list-group-item d-flex justify-content-between p-2">
      <span aria-label="span" className="align-self-center" onClick={() => onDoneTodo(todo.id)}>
        {todo.done ? <s>{todo.description}</s> : todo.description}
      </span>
      <button className="btn btn-danger" onClick={() => onDeleteTodo(todo.id)}>
        Borrar
      </button>
    </li>
  );
};
