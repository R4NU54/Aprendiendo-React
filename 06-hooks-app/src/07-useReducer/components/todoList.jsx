import { TodoItem } from './todoItem';

export const TodoList = ({ todos = [], onDeleteTodo, onDoneTodo }) => {
  return (
    <ul className="d-flex gap-2 list-group">
      {todos.map(todo => (
        <TodoItem key={todo.id} todo={todo} onDeleteTodo={onDeleteTodo} onDoneTodo={onDoneTodo} />
      ))}
    </ul>
  );
};
