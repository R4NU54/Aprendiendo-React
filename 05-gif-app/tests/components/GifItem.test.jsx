import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas en el componente GifItem', () => {
  const title = 'A title';
  const url = 'https://localhost/algo.jpg';

  const component = (
    <GifItem
      title={title}
      url={url}
    />
  );

  test('Debe hacer match con el snapshot', () => {
    const { container } = render(component);

    expect(container).toMatchSnapshot();
  });

  test('debe de mostrar la imagen con el Url y el Alt indicado', () => {
    render(component);

    const { src, alt } = document.querySelector('img');

    expect(src).toBe(url);
    expect(alt).toBe(alt);
  });

  test('debe de mostrar el titulo en el componente  ', () => {
    render(component);
    expect(screen.getAllByText(title)).toBeTruthy();
  });
});
