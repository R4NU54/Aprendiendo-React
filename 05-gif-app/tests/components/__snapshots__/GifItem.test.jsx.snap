// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="A title"
      src="https://localhost/algo.jpg"
    />
    <p>
      A title
    </p>
  </div>
</div>
`;
